crates/bevy_asset/src/io/wasm/opfs.rs

```diff
@@ -0,0 +1,735 @@
+        let shadow_root = self.shadow_root().await?;
+        let canon = Self::canonical(path)?;
+        let _ = get_file(&shadow_root, canon.iter().copied(), false, false).await?;
+
+        let [parent @ .., file] = canon.as_slice() else {
+            unreachable!("path valid based on above guard");
+        };
+
+        let parent_handle = get_directory(&shadow_root, parent.iter().copied(), false).await?;
+
+        remove_entry(&parent_handle, file).await?;
+
+        Ok(())
+    }
+
+    async fn remove_meta<'a>(&'a self, path: &'a Path) -> Result<(), AssetWriterError> {
+        self.remove(&get_meta_path(path)).await
+    }
+
+    async fn rename<'a>(
+        &'a self,
+        old_path: &'a Path,
+        new_path: &'a Path,
+    ) -> Result<(), AssetWriterError> {
+        let mut buffer = Vec::new();
+
+        self.read(old_path)
+            .await
+            .map_err(|error| {
+                AssetWriterError::from(std::io::Error::new(std::io::ErrorKind::Other, error))
+            })?
+            .read_to_end(&mut buffer)
+            .await?;
+        self.write(new_path).await?.write(&buffer).await?;
+        self.remove(old_path).await?;
+
+        Ok(())
+    }
+
+    async fn rename_meta<'a>(
+        &'a self,
+        old_path: &'a Path,
+        new_path: &'a Path,
+    ) -> Result<(), AssetWriterError> {
+        self.rename(&get_meta_path(old_path), &get_meta_path(new_path))
+            .await
+    }
+
+    async fn remove_directory<'a>(&'a self, path: &'a Path) -> Result<(), AssetWriterError> {
+        let shadow_root = self.shadow_root().await?;
+        let canon = Self::canonical(path)?;
+        let _ = get_directory(&shadow_root, canon.iter().copied(), true).await?;
+
+        let [parent @ .., directory] = canon.as_slice() else {
+            unreachable!("path valid based on above guard");
+        };
+
+        let parent_handle = get_directory(&shadow_root, parent.iter().copied(), false).await?;
+
+        remove_entry(&parent_handle, directory).await?;
+
+        Ok(())
+    }
+
+    async fn remove_empty_directory<'a>(&'a self, path: &'a Path) -> Result<(), AssetWriterError> {
+        let shadow_root = self.shadow_root().await?;
+        let canon = Self::canonical(path)?;
+        let handle = get_directory(&shadow_root, canon, true).await?;
+        let mut stream = get_entries(&handle).await;
+
+        if stream.next().await.is_some() {
+            return Err(AssetWriterError::from(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "Directory is not empty",
+            )));
+        }
+
+        self.remove_directory(path).await
+    }
+
+    async fn remove_assets_in_directory<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> Result<(), AssetWriterError> {
+        let shadow_root = self.shadow_root().await?;
+        let handle = get_directory(&shadow_root, Self::canonical(path)?, true).await?;
```

Comment:
```suggestion
        let handle = get_directory(&shadow_root, Self::canonical(path)?, false).await?;
```

Probably shouldn't be creating directories to remove assets.